### PR TITLE
Add distclean rule to makefile

### PR DESCRIPTION
**Pull request by AngheloAlf**
Adds a ~`clean_everything`~ `distclean` that aims to clean all generated files not deleted by `clean`.

It does:
- Delete every file that `make clean` deletes.
- Delete every extracted asset in `assets/`. It doesn't deletes the `assets/xml/` folder.
- Delete `baserom/`
- Delete the generated/compiled tools and files in `tools/`. It only invokes the ~`clean_everything`~ `distclean` rule of the tool's makefile.
  - This will also delete every file generated when compiling ZAPD.

I did this because sometimes your local copy can break because of an residual file or something, and just running `make clean && make setup -j && make -j` will not fix the issue, but erasing every generated file may fix it.

Let me know if I forgot any generated file.

~PS: For comedic purposes, the rule `nuke` was also added and it does the same as ~`clean_everything`~ `distclean`, so you can do `make nuke` instead if you prefer.~

### CODE REVIEW

**Review by louist103 — APPROVED** [review]

**Comment by fig02** [comment]
imo id name the rule `clean_all` to make it easier to type, and remove "nuke". feels like having an alias for the same thing could cause confusion and doesnt seem necessary

**Comment by fig02** [comment]
ah i see its since been changed. still feel the same about the alias though

File: Makefile
Comment (Roman971) [inline Makefile:198]: Shouldn't this be `ASSET_BIN_DIRS` instead of `TEXTURE_BIN_DIRS`?

File: Makefile
Comment (Roman971) [inline Makefile:203]: I agree with Fig that this alias seems unnecessary

File: Makefile
Comment (Roman971) [reply Makefile:198]: Looking at the rest of the makefile a bit closer, it seems like `TEXTURE_BIN_DIRS` is just a duplicate of `ASSET_BIN_DIRS`, so we should probably remove it and use `ASSET_BIN_DIRS` instead.
On a similar note, `TEXTURE_DIRS` seems useless and looks like it could be removed entirely.

Comment (Roman971) [reply Makefile:198]: I can make a follow up PR after we merge this if you prefer (to clean up what I just mentioned)

Comment (AngheloAlf) [reply Makefile]: I don't have problems doing that cleanup myself, I can do it if you like.

File: Makefile
Comment (AngheloAlf) [reply Makefile]: Removed.

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — COMMENTED** [review]
The `TEXTURE_DIRS` declaration itself can also be removed from the file, but otherwise this looks good to me

**Review by Roman971 — APPROVED** [review]

**Pull request merged**
